pkg/github/projects/project_items.go

```diff
@@ -0,0 +1,318 @@
+	for _, f := range p.Fields {
+		if exclude[f.Common.DataType] {
+			continue
+		}
+		fieldType := fieldTypes[f.Common.DataType]
+		field := data.NewField(f.Common.Name, nil, fieldType)
+		frame.Fields = append(frame.Fields, field)
+		fields = append(fields, f)
+	}
+
+	// get the values for each item and append to the frame
+	for _, v := range p.Items {
+		var vals []any
+
+		vals = append(vals, v.ID)
+		vals = append(vals, v.IsArchived)
+		vals = append(vals, v.Type)
+		vals = append(vals, v.UpdatedAt.Time)
+		vals = append(vals, v.CreatedAt.Time)
+		vals = append(vals, closedDate(v.Content))
+
+		// create a lookup of field names to values
+		fieldValue := map[string]any{}
+		fieldValue["type"] = v.Type
+		fieldValue["created_at"] = v.CreatedAt.Time
+		fieldValue["closed_at"] = closedDate(v.Content)
+		// populate some field values from content
+		// we could get them from fieldValues but content has explicit types
+		fieldValue["Assignees"] = getAssignees(v.Content)
+		fieldValue["Milestone"] = milestone(v.Content)
+
+		for _, fv := range v.FieldValues.Nodes {
+			name, val := nameValue(fv)
+			fieldValue[name] = val
+		}
+
+		// add the values to an array that we will append to the frame row
+		for _, f := range fields {
+			val := fieldValue[f.Common.Name]
+			vals = append(vals, val)
+		}
+
+		// if there are no filters, just append the values
+		if len(p.Filters) == 0 {
+			frame.AppendRow(vals...)
+			continue
+		}
+
+		// check for a filter match and append to the frame if there is a match
+		match := filter(fieldValue, p.Filters)
+		if match {
+			frame.AppendRow(vals...)
+		}
+		continue
+
+	}
+
+	return data.Frames{frame}
+}
+
+// get the field name and value from the response model
+func nameValue(fv FieldValue) (string, any) {
+	// DateValue, SelectValue, TextValue etc all have the field data type
+	dataType := fv.DateValue.Field.Common.DataType
+	switch dataType {
+	case "DATE":
+		return fv.DateValue.Field.Common.Name, date(fv)
+	case "SINGLE_SELECT":
+		return fv.SelectValue.Field.Common.Name, fv.SelectValue.Name
+	case "TEXT", "TITLE":
+		return fv.TextValue.Field.Common.Name, fv.TextValue.Text
+	case "ITERATION":
+		return fv.IterationValue.Field.Common.Name, fv.IterationValue.Title
+	case "LABELS":
+		return fv.LabelsValue.Field.Common.Name, labels(fv)
+	case "NUMBER":
+		return fv.NumberValue.Field.Common.Name, fv.NumberValue.Number
+	case "REVIEWERS":
+		return fv.ReviewerValue.Field.Common.Name, reviewers(fv)
+	case "REPOSITORY":
+		return fv.RepoValue.Field.Common.Name, &fv.RepoValue.Repository.Name
+	}
+	return fv.DateValue.Field.Common.Name, nil
+}
+
+// GetAllProjectItems uses the graphql endpoint API to list all project items in the repository
+func GetAllProjectItems(ctx context.Context, client models.Client, opts models.ProjectOptions) (*ProjectItemsWithFields, error) {
+	if opts.Kind == 0 {
+		projects, err := getAllProjectItemsByOrg(ctx, client, opts)
+		projects.Filters = opts.Filters
+		return projects, err
```

Comment:
Will projects be nil if getAllProjectItemsByOrg returns an err? if not I think we should `return nil, err` like below.